Signing image builds (#59)
openshift-psap/special-resource-operator: docs/debug.md

# How to debug recipes

The Special Resource Operatore (SRO) can read the states either from the local,
http or a ConfigMap resource.

To create the states in the template directory as a CM we can run the
following:

```bash

> VERSION=0.0.1 REPO=exammple SPECIALRESOURCE=multi-build make assets

Reviewer: ```suggestion
VERSION=0.0.1 REPO=example SPECIALRESOURCE=multi-build make assets
```

```

> This command will create a CM with all the states and SRO will use them in the

Reviewer 2: is this only for development or can be used in production also?
as the partner may not want to push the is assets into the SRO repo

next reconcilation loop. This makes it easy to override manifests during
development without rebuilding SRO.

To update the CR one can run:

```bash

> VERSION=0.0.1 REPO=exammple SPECIALRESOURCE=multi-build make

Reviewer: ```suggestion
VERSION=0.0.1 REPO=example SPECIALRESOURCE=multi-build make
```

```

For this to work one has to create a `kustomization.yaml` that will be processed

> during a make run. One cannot just override one state all states are either
> read from one resource. Here is an `kustomization.yaml` as an example from the

Reviewer: ```suggestion
during a make run. One cannot just override one state, all states are
read from one source (http, local, or ConfigMap). Here is an `kustomization.yaml` as an example from the
```
Found this sentence confusing at first.

`multi-build` recipe.

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

generatorOptions:
  disableNameSuffixHash: true

configMapGenerator:
- files:
  - templates/0000-buildconfig.yaml
  - templates/1000-buildconfig.yaml
  - templates/2000-imagesign.yaml
  name: multi-build
namespace: multi-build
```

Another field was added to the CR, namely `debug` that can be set to true to get
all the manifests, hooks and values printed on the console. Can be valuable for
verifying if all `Values` are set and correclty interpreted.

```yaml
apiVersion: sro.openshift.io/v1beta1
kind: SpecialResource
metadata:
  name: multi-build
spec:

>   debug: false

Reviewer 2: I think this needs to be true in this example
